File: GeneratedEntityReaderClasses/Components/HomingComponent.cs
namespace GeneratedNoitaClasses;
public class HomingComponent : NoitaComponentBase,  iNoitaType<HomingComponent>{
	public NoitaBool Deleted { get; set; } = new();
	public NoitaBool Enabled { get; set; } = new();
	public NoitaString Tags { get; set; } = new();
	public NoitaString target_tag { get; set; } = new();
	public NoitaBool target_who_shot { get; set; } = new();
	public NoitaFloat detect_distance { get; set; } = new();
	public NoitaFloat homing_velocity_multiplier { get; set; } = new();
	public NoitaFloat homing_targeting_coeff { get; set; } = new();
	public NoitaBool just_rotate_towards_target { get; set; } = new();
	public NoitaFloat max_turn_rate { get; set; } = new();
	public NoitaUInt predefined_target { get; set; } = new();
	public NoitaBool look_for_root_entities_only { get; set; } = new();

    public void Read(NoitaStream s){
		Deleted.Read(s);
		Enabled.Read(s);
		Tags.Read(s);
		target_tag.Read(s);
		target_who_shot.Read(s);
		detect_distance.Read(s);
		homing_velocity_multiplier.Read(s);
		homing_targeting_coeff.Read(s);
		just_rotate_towards_target.Read(s);
		max_turn_rate.Read(s);
		predefined_target.Read(s);
		look_for_root_entities_only.Read(s);

    }
    public void Write(NoitaStream s){
		Deleted.Write(s);
		Enabled.Write(s);
		Tags.Write(s);
		target_tag.Write(s);
		target_who_shot.Write(s);
		detect_distance.Write(s);
		homing_velocity_multiplier.Write(s);
		homing_targeting_coeff.Write(s);
		just_rotate_towards_target.Write(s);
		max_turn_rate.Write(s);
		predefined_target.Write(s);
		look_for_root_entities_only.Write(s);

    }
}

File: GeneratedEntityReaderClassesPython/GeneratedClasses/Components/HomingComponent.py
from dataclasses import dataclass, field
from BasicNoitaTypes import NoitaBool
from BasicNoitaTypes import NoitaString
from BasicNoitaTypes import NoitaFloat
from BasicNoitaTypes import NoitaUInt

from NoitaStream import NoitaStream
@dataclass
class HomingComponent:
    Deleted:NoitaBool = field(default_factory=NoitaBool)
    Enabled:NoitaBool = field(default_factory=NoitaBool)
    Tags:NoitaString = field(default_factory=NoitaString)
    target_tag:NoitaString = field(default_factory=NoitaString)
    target_who_shot:NoitaBool = field(default_factory=NoitaBool)
    detect_distance:NoitaFloat = field(default_factory=NoitaFloat)
    homing_velocity_multiplier:NoitaFloat = field(default_factory=NoitaFloat)
    homing_targeting_coeff:NoitaFloat = field(default_factory=NoitaFloat)
    just_rotate_towards_target:NoitaBool = field(default_factory=NoitaBool)
    max_turn_rate:NoitaFloat = field(default_factory=NoitaFloat)
    predefined_target:NoitaUInt = field(default_factory=NoitaUInt)
    look_for_root_entities_only:NoitaBool = field(default_factory=NoitaBool)
    
    def Read(self, s:NoitaStream):
        self.Deleted.Read(s)
        self.Enabled.Read(s)
        self.Tags.Read(s)
        self.target_tag.Read(s)
        self.target_who_shot.Read(s)
        self.detect_distance.Read(s)
        self.homing_velocity_multiplier.Read(s)
        self.homing_targeting_coeff.Read(s)
        self.just_rotate_towards_target.Read(s)
        self.max_turn_rate.Read(s)
        self.predefined_target.Read(s)
        self.look_for_root_entities_only.Read(s)
    
    def Write(self, s:NoitaStream):
        self.Deleted.Write(s)
        self.Enabled.Write(s)
        self.Tags.Write(s)
        self.target_tag.Write(s)
        self.target_who_shot.Write(s)
        self.detect_distance.Write(s)
        self.homing_velocity_multiplier.Write(s)
        self.homing_targeting_coeff.Write(s)
        self.just_rotate_towards_target.Write(s)
        self.max_turn_rate.Write(s)
        self.predefined_target.Write(s)
        self.look_for_root_entities_only.Write(s)

